### DISCUSSION TITLE
Dev setup

### DISCUSSION THREAD

**Discussion by wormstein** [EdNekebno/DGTCentaurMods#16] [General]
Hi,
just want to start a topic on best practices for dev serup.

My dev serup is something like this:
- pycharm on my desktop
- board file system mapped on my desktop using sshfs
- project loaded in pycharm
- testing using python terminal to the board
- symlink of project to /home/pi/centaur/py

How is yours?

-Adrian

**Comment by EdNekebno** [comment]
He he...we might end up just proving I'm not very good when it comes to development best practice. But pretty similar.

Also pycharm on my desktop. I have Samba installed on the centaur. My files are in /home/pi/centaur/py and I have it set up in samba as a share. So when I want I can start samba and map a drive. (I like the sshfs idea more though). For testing things I then ssh in and run python from the command line.

I'm a bit old school sometimes and hacky/inclined to test things. So it's not unusual, if I know I won't be keeping a file, for me to ssh in and put something small and badly coded together with vi to test. For some reason old school me finds that quicker/simpler. I often try to get the world record for the most ssh windows open at one time. That was, after all, why 4k monitors were invented :)

**Comment by wormstein** [comment]
Well, don't imagine I'm a coding guru as I come from a sys admin background... but I am passionate about chess and I love hacking devices. So this project fits all of them.
And you're right... for the size of the project pycharm is too much for now. I do often find myself doing changes in vim which I actually tuned as a pretty good IDE.

-Adrian
